it/controllers/incomeSources/cease/CheckCeaseIncomeSourceDetailsControllerISpec.scala

```diff
@@ -0,0 +1,226 @@
+  val pageTitleMsgKeyUK = messagesAPI("incomeSources.ceaseUKProperty.checkDetails.heading")
+  val redirectUriUK = controllers.incomeSources.cease.routes.IncomeSourceCeasedObligationsController.show(UkProperty).url
+  val requestUK: UpdateIncomeSourceRequestModel = UpdateIncomeSourceRequestModel(
+    nino = testNino,
+    incomeSourceId = ukProperty.incomeSourceId,
+    cessation = Some(Cessation(true, Some(LocalDate.parse(cessationDate))))
+  )
+
+  val sessionCeaseForeignPropertyEndDate = Map(forms.utils.SessionKeys.ceaseForeignPropertyEndDate -> cessationDate)
+  val showCheckCeaseForeignPropertyDetailsControllerUrl = controllers.incomeSources.cease.routes.CeaseCheckIncomeSourceDetailsController.show(ForeignProperty).url
+  val formActionFP = controllers.incomeSources.cease.routes.CeaseCheckIncomeSourceDetailsController.submit(ForeignProperty).url
+  val pageTitleMsgKeyFP = messagesAPI("check-cease-foreign-property-details.heading")
+  val redirectUriFP = controllers.incomeSources.cease.routes.IncomeSourceCeasedObligationsController.show(ForeignProperty).url
+  val requestFP: UpdateIncomeSourceRequestModel = UpdateIncomeSourceRequestModel(
+    nino = testNino,
+    incomeSourceId = foreignProperty.incomeSourceId,
+    cessation = Some(Cessation(true, Some(LocalDate.parse(cessationDate))))
+  )
+
+  s"calling GET ${showCheckCeaseBusinessDetailsControllerUrl}" should {
+    "render the Cease Business Details Page" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with multiple business and property")
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, businessOnlyResponse)
+        When(s"I call GET ${showCheckCeaseBusinessDetailsControllerUrl}")
+        val res = IncomeTaxViewChangeFrontend.getCheckCeaseBusinessDetails(sessionCeaseBusinessDetails)
+        verifyIncomeSourceDetailsCall(testMtditid)
+
+        res should have(
+          httpStatus(OK),
+          pageTitleIndividual(pageTitleMsgKey),
+          elementTextBySelectorList(".govuk-summary-list__key", "dt:nth-of-type(1)")(messagesAPI("incomeSources.ceaseBusiness.checkDetails.dateStopped")),
+          elementTextBySelectorList(".govuk-summary-list__value", "dd:nth-of-type(1)")(businessEndShortLongDate),
+          elementTextByID("change-link")(changeLink),
+
+          elementTextBySelector("dl:nth-of-type(1) div:nth-of-type(2) dt:nth-of-type(1)")(messagesAPI("incomeSources.ceaseBusiness.checkDetails.businessName")),
+          elementTextBySelector("dl:nth-of-type(1) div:nth-of-type(2) dd:nth-of-type(1)")(testBusinessName),
+          elementAttributeBySelector("form", "action")(formActionSE)
+        )
+      }
+    }
+
+    "render the Cease Business Page with unknown address and title" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with multiple business and property")
```

Comment:
Same here, it's a single business response (but you might want to add that it's not got an address)